### PR TITLE
replace gettext and intltool with Babel

### PR DESCRIPTION
**Pull request by mattrose**
I've managed to replace all of the gettext and intltool external dependency and replace it with the python babel library instead.  It's actually much simpler as well, which is nice.

Notes:

To generate the `po/terminator.pot` you can now just run `python setup.py extract_messages`, I've replaced the contents of `po/genpot.sh` with the correct command
It uses the same mechanism to generate the .mo files tied into the same `python setup.py build` command.  

The old `build` command would run a tool called "intltool-merge" that would merge all of the translated strings into the `data/terminator.desktop` and `data/terminator.appdata.xml`. The new version just pulls the translations out and uses the same localization machanisms as the python files  

Notes for distributors:  If your build environment is not able to run pip install or is otherwise detached from the net, you will need to add the babel library package explicitly to the BuildRequires section of the respective package.  In fedora it's python3-babel.

I've run a few tests and make sure it all works but I would appreciate it if other people would take a look too.  @lazyfrosch can merge it once he thinks it's ready.

### CODE REVIEW

File: data/terminator.metainfo.xml [hunk 1]
```diff
@@ -2,47 +2,47 @@
...
   </screenshot>
   <screenshot>
    <image>http://4.bp.blogspot.com/-rRxALSpEEZw/Vdeu58JgpnI/AAAAAAAAA9o/XewWKJ5HNo4/s1600/terminator_main_complex.png</image>
-   <_caption>Getting a little crazy with the terminals</_caption>
+   <caption translatable="yes">Getting a little crazy with the terminals</caption>
   </screenshot>
   <screenshot>
-   <image>http://2.bp.blogspot.com/-t_8oRyMXUls/VdemmRVnZnI/AAAAAAAAA88/rHIr8L1X7Ho/s1600/terminator_prefs_global.png</image>
-   <_caption>The preferences window where you can change the defaults</_caption>
+   <image>http://2.bp.blogspot.com/-t_8oRyMXUls/VdemmRVnZnI/AAAAAAAAA88/rHIr8L1X7Ho/s1600/terminatorp translatable="yes"refs_global.png</image>
```
Comment (JakubVanek) [inline data/terminator.metainfo.xml:44]: The `` got corrupted by a replacement

File: po/genpot.sh [hunk 2]
```diff
@@ -1,11 +1,4 @@
...
-ln -s remotinator ../remotinator.py
-
-# Make translation files
-intltool-update -g terminator -o terminator.pot -p
-
-# Cleanup after stupid workaround
-rm ../terminator.py
-rm ../remotinator.py
+cd ..
+python setup.py extract_messages
```
Comment (lazyfrosch) [inline po/genpot.sh:3]: Please drop this file, we should add the setup.py call to developer docs, e.g. `DEVELOPMENT.md`

File: data/terminator.metainfo.xml (hunk 1 above)
Comment (lazyfrosch) [reply data/terminator.metainfo.xml:44]: What Jakub said :smile:

File: po/terminator.pot [hunk 3]
```diff
@@ -1,1776 +0,0 @@
-# SOME DESCRIPTIVE TITLE.
```
Comment (lazyfrosch) [inline po/terminator.pot:1]: Why deleting the POT? Shouldn't we keep it for translators?

File: data/terminator.metainfo.xml (hunk 1 above)
Comment (dmaphy) [reply data/terminator.metainfo.xml:44]:
```suggestion
   <image>http://2.bp.blogspot.com/-t_8oRyMXUls/VdemmRVnZnI/AAAAAAAAA88/rHIr8L1X7Ho/s1600/terminatorprefs_global.png</image>
```

**Comment by lazyfrosch** [comment]
I'm still fiddling around with it.

File: po/terminator.pot (hunk 3 above)
Comment (mattrose) [reply po/terminator.pot:1]: Hmm, definitely shouldn't have deleted that.

File: po/genpot.sh (hunk 2 above)
Comment (mattrose) [reply po/genpot.sh:3]: Sure.  Maybe a TRANSLATIONS.md file?  Does anyone have any best practices here?  Maybe I'll tag it on to the bottom of INSTALL.md for now.

File: po/terminator.pot (hunk 3 above)
Comment (mattrose) [reply po/terminator.pot:1]: Looks like you brought it back in a following commit.

Comment (lazyfrosch) [reply po/terminator.pot:1]: Yes, checkout, commit, update. :wink:

File: data/terminator.metainfo.xml [hunk 4]
```diff
@@ -2,47 +2,47 @@
...
   </screenshot>
   <screenshot>
    <image>http://4.bp.blogspot.com/-rRxALSpEEZw/Vdeu58JgpnI/AAAAAAAAA9o/XewWKJ5HNo4/s1600/terminator_main_complex.png</image>
-   <_caption>Getting a little crazy with the terminals</_caption>
+   <caption translatable="yes">Getting a little crazy with the terminals</caption>
   </screenshot>
   <screenshot>
-   <image>http://2.bp.blogspot.com/-t_8oRyMXUls/VdemmRVnZnI/AAAAAAAAA88/rHIr8L1X7Ho/s1600/terminator_prefs_global.png</image>
-   <_caption>The preferences window where you can change the defaults</_caption>
+   <image>http://2.bp.blogspot.com/-t_8oRyMXUls/VdemmRVnZnI/AAAAAAAAA88/rHIr8L1X7Ho/s1600/terminatorprefs_global.png</image>
```
[the remark this answers is not in the document]
Comment (mattrose) [reply data/terminator.metainfo.xml:44]: looks like @lazyfrosch and @dmaphy fixed this in d6e4653

[the remark this answers is not in the document]
Comment (lazyfrosch) [reply data/terminator.metainfo.xml:44]: Kind of, `terminatorprefs` is still wrong, but doesn't matter for the link, and I will move the links away from blogspot anyways.

**Comment by lazyfrosch** [comment]
@mattrose I refactored the code a bit, and added a general update command.

**Comment by lazyfrosch** [comment]
Thanks @mattrose for your work!

**Comment by lazyfrosch** [comment]
This will be reverted in #100

**Pull request merged**